### ISSUE TITLE
The Squire Support Gems Sorting

### ISSUE THREAD

**Issue by Phelnax** [PathOfBuildingCommunity/PathOfBuilding#3508]
### Check version

- [X] I'm running the latest version of Path of Building and I've verified this by checking the [changelog](https://github.com/PathOfBuildingCommunity/PathOfBuilding/blob/master/CHANGELOG.md)

### Check for duplicates

- [X] I've checked for duplicate issues by using the search function of the [issue tracker](https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues)

### What is the expected behaviour/value?

When having an active skill in a socket group in weapon 1 and adding a support to a socket group in weapon 2 with the squire equipped, sorting should sort support gems with them properly supporting the skill socketed in weapon 1.

### What is the actual behaviour/value?

When having an active skill in a socket group in weapon 1 and adding a support to a socket group in weapon 2 with the squire equipped, the DPS calculations work properly but the list is sorted as if the support gems did not support the skill.

### How to reproduce the issue

Equip a 1h main hand.
Equip the squire.
Put an active skill socketed in weapon 1.
Make that active skill the main skill.
Create a new socket group that is socketed in weapon 2.
Attempt to add a gem to the new socket group and look at list.

### Build code

```shell
eNrFW21z2kgS_hx-xRRVucpWCtDoDcln7xZ-p84kXnCSu0-psTSA1kJDpBEOu7X__XpGwggbTKPk6uIUFtLT7z093aPk-Lfvs5gseJpFIjlp0rbRJDwJRBglk5Pmp7vLltf87dfG8S2T04_j0zyK1ZNfG2-O9TWJ-YLHQNckkqUTLj-vOFlfgdOcJXLKRTJgf4j0SoQnzQ8i4U1yz5IwkqtvQcyy7AOb8ZPmKADiJmFZwJPwbH2_AM5YlIxE8MDlVSryuRa7iPjjQISA6Q9uPw7vKkKjpCoUdH5zfBuzJU9HkkmSwcdJswemswm_jiSwYnEOfKjpuG3fgl_NzqtE52wGnwfQjeach2t82zY9p0vLz_8_0W3KL8ZjHshowc_SSJ5NWRKs7eu2dxp2OHqQxzKaxxFPK1rupLh-wZ4axi7wnZAsPr8drbFd32m71DW0B9z_DZ1YJ9BOzb5EcnoaQ2RqSOlPkkjyKqFlQLZ1nT0Emznqm23vlQRVCr4URD3TbTuO5byuoSK-FVEmkh9w4gEGnonZfZQ8cyZO1RVpbzE5VNEBS9iZyNbRNneup5tozDegOxPjYoTDDSGWOKRS85anUEYljkApexDBSK6XrrlbY_6tCvR2RvOcf8exqwJ3s-snEseuCqSOv1u9hZB6b8I48gnlGq9U6OBIYftJgGP6KUl5xtNFpbC7SIIysJjqqQiHfMKRlt5wHkyvYF8eMslxafmEcl53jsKinKOAW5zjIAkOcI4ifOactv8a-ED3fGFpiKgWCU8ny9E04jECrX1ZJUH5tEqATIcqyYF2XyxYVl1cluO9bk6Bx2UHhx0XCEL-rIVwvZ9IcpuKP1TnFB9I10tnIl_XR8v0XzW8gKPsvp0usyiATVV3AEMe5sFGAfN3R2PVBA7Egs9gaehuEprsatZ3d1GfxtChP7Pf7L5iVRxvI9mpXU9KFjyci3DCsSRayOEUo3w-hyKhMg1Ldxml4OosqmygLReB_ggTxRmb7xdwJtR6xApYo9ECbqLJVCYw4-GlPCPB2zJlIjvAmDUcLeIyjzda-t0Bf4g2kZLdx_yIGN-pRy2bGbtXvngEZaZq-s0Qq7KCHrB1J2PttCDlyZ9LNP8NOErARRLmqcputIznFC_FHHf06YC66s_mIpX65hmLg0yz7CfzXJJEj_aZ8vzXJJ_dq1mw-K0m-84L5CzKgq_3-Xis5v0mKJPqQ4qLy8uLs7v-54uSZMR1oSOBiGM2z3h40kyiuEmisAzzCMp7IBFoGDvL-Wk_FrbuII3AffuhalZEqKpH-P04NU0jYBAHFnOsi-6Wc64inuG8VJbH_Vg9_SJwxRyJMUtPqxin84AtEbinzgWBvYh5L4rVDonx0wDSt9haMWjY5tLoPpeojNIzB0ID1X0jzKr2l6jMx8DKTgchvmwG9yPLvQOTdtBgIEN0zscckhm1kHVxuGMPPCnL3qrEHetVlJEMat8Vn2WnS9hbLpVDq-dMG09hy0lzKGshH7M8Vvd_z1kcyaUeYCu3b4ojV5MCg6l4VF1KwUct2gwcfXNTPOnFsmShhJT8C6OUdvo4tacbPf1d667PVKMkiPMQJtNy-yxNjtm9EtwkE3X6eibyRJZPeKJ2y3BlQhYLNcxwNgfH0eZzQUqGUuPNMehV0l7F4p7F5krS2sTqY_okQLHp66SCTvhbYWRfr3LtpRJxy1KpDZrwmXo64JKFTLJOX4JDOsorHa0RXFU5FR5_btSK49pPQeECuFIblGrQS400dhAl-iR8lM9mIoE5E4qkZNmsTKxC8t5wFMn2MwNibgmIIioyuEhcdXmXck5YgdKm0TJ54Ev1hF551lA8M8nTZVnhlFYJbNNw4XieZTWJBHaVtwPULQ_-C_IiHy5CKObhB0XXKe58Gt7oizdTKefZUafz-PjYnjM5FWP-HaatdiBmnTmwAS1b2u0tJajTgz-n6qN3cqIZdVacjos3CFnBtvymdVWaUNekHjhcFhlT9kBbcJQ6BgLnUs-yMTjb8SwEzjR9H8PP9tyuj-FnUYqA-baB8YrluihrLdvGWUt9D8PPtH2MFZbVdTBiLeri1HO7poPAOQ71Mcli-4aHwVGLdjG4ru-ixLqeZ6JwYAfKza6FCYdt2IaBCoeBEUtNl1ooMywfFV3IFhcTXd_xMFlAbR8l1-_6XYx6nt3FiDX9LjVRyUx9F1XRLMNB4UzPQ1W0ro2qLZbv-Si5rouxg3ZNiquQ6BrUxe0INkquDfmMSReHQpHE2Gs4FiYPqGdbqLQHBbuoouZ6uB2r66LsMKmLqi4UGGJwNnVReWVaqPAaXctBbZU-qki6PkXVSLO7LVugn1t3O_AFWjfd5aneTl18EDBgqmfq5uqL7vw-R_yRZJylwXQkU9XO_SnE7D-61VPDxTVncsDmZb-pnpVziVWOJZLJ8wiawVRPU6tOVAH_vdLvWPfhZY-prke8aKfzjMPcJZKwaFn1bSWnOBICoB6_zKJrHLIUevYj8ulD__dPF427KSejb3mU8gbM5xPouckQOuOQFANt44azSc6PyAi693TCG-WIdERMo1E664hcteCnoS0a8m9HpGs0-rN5HAWRekob1DTekn4SpJzBSEj0fEmGoPICWuDGNcuIRUpmjR709uU1YSknX6YQpcZ7k0hBtAQixiUAWJXjHFHDSePp7geRtC6-B3yuZlEWb6BIwBLC4kw83S2Hz3EqZmQJVpIB9PvkmiVhgxrmWzVOlIoXQzmBJ6QcuxvvrbekOM9RCmrLdO4ot6_9vwCnMzVumMUobG2NRe_-nqfQrP8Dopj9MyPXIk0aV4LJjRufC15H5Ba8Y7attrG-dZankECyUZzZgcZPT8yXjLaG8rQFP5VQutVImo2_SkuO6N9_pSyZ8COj7fz9zm9R85eqp_QbAVJM_Gsic4OIGi1qv0JVgZpGyzI2oOplwBZkLwwz8s5uub-ocLzzlF5VrLqrpWQb_G2j5W7yL0Zooo4L1WshAqs6euBlqBvvqUHUKRJRR1-AhjnxX5BEUTIhxREbJEjCZxHPGqbzlqTqdRLcK56R8xyirE5J4O8KpqrPQIRDpRPRmqn3Rk6RLnRVn3YizL0Iay_C3otY_fuBdXaPYFQuz5vXBxirsro6RH8Jquys5urouihoO4tZ6YhyKH0pdvTI5qR3v8wytiogZMvOso30OZmJI3th6AbqVAioYPUU2mbLHqVOeSwPJLmMWfZALJRK5jaVrHruxTnB_Elkdj0l3XrS3PrepHsCvC2f9kTgmsczLg8VtCWT3Dpy9qQfVNiFen92OGOr_iqideTZ9XKB1ss8e394MI49lO229KppOKruODX0qVEDrX1iwuWqrTssfNu9XCu_nDpEqCWJCUQtO53abjVrxN2tt5KseulrHZ5ldv2yXyOnaW3nW5iEsDGgGmr_QPODjOSLrhIdBbdGVtYPww-kC7YkK1rkqsFDa3huqGYis06JMWu7t2a5wMezjjkWwk-0fhjs2t5y6liDWS_1k3yPSr1ZHm_Z24qxFAZGfWanD8v0v6MSyTiaAOK48_y_Zf0XiAxIgQ==
```

**Comment by onefuncman** [comment]
similar bug where doesn't calc damage for 3rd gem in squire unless it is linked, despite in game showing gem support indicator icon
https://pobb.in/cuLeDIWBMYj5
![image](https://github.com/PathOfBuildingCommunity/PathOfBuilding/assets/45834012/b03c20b5-2a30-4c35-8fb6-d9047748b335)

**Comment by Paliak** [comment]
Possibly fixed by #6707 ?

**Comment by Paliak** [comment]
> How to reproduce the issue
> 
> Equip a 1h main hand.
> Equip the squire.
> Put an active skill socketed in weapon 1.
> Make that active skill the main skill.
> Create a new socket group that is socketed in weapon 2.
> Attempt to add a gem to the new socket group and look at list.

This should work correctly now.
![obraz](https://github.com/PathOfBuildingCommunity/PathOfBuilding/assets/91493239/5be196a6-e19f-4cfb-8a72-41e9ade34686)
